[Kiosk SDK] Adds support for attaching rules to Transfer Policy (#12945)
MystenLabs/sui: sdk/kiosk/README.md

Here are some indicative examples on how to use the kiosk SDK.

<details>

> <summary>Find an addresses' owned kiosk(s)</summary>

Reviewer: "Find an address" or "find addresses"
But maybe more clear to say "Find the kiosks owned by an address" or "Find the kiosks an address owns"

```typescript
import { getOwnedKiosks } from '@mysten/kiosk';
import { Connection, JsonRpcProvider } from '@mysten/sui.js';

const provider = new JsonRpcProvider(
	new Connection({ fullnode: 'https://fullnode.testnet.sui.io:443' }),
);

// You could use these to fetch the contents for each kiosk, or use the `kioskOwnerCap` data for other actions.
const getUserKiosks = async () => {
	const address = `0xAddress`;
	const { data } = await getOwnedKiosks(provider, address);
	console.log(data); // kioskOwnerCaps:[], kioskIds: []
};
```

</details>

<details>

> <summary>Getting the listings & items by the kiosk's id</summary>

Reviewer: just kiosk ID or use the label used in code and format as code inline

```

</details>

<details>

<summary>Create a Transfer Policy for a Type</summary>

You can only create a TransferPolicy for packages that you own the `publisher` Object.

It's recommended (unless you have a more advanced use case) to create only one transfer policy per type, or make sure that all transfer policies have the same rules.

If you have multiple transfer policies, someone could purchase an item of that Type from a kiosk by using the easiest (to resolve) rules.

> You could extend the following snippet to do a check before creating the transfer policy, by using the `queryTransferPolicy` function, similar to the `purchaseAndResolvePolicies` example.

Reviewer: As a best practice, you should use a single Transfer policy per type (T). If you use more than one policy for a type, and the rules for each differ, anyone that meets the conditions for any of the attached policies can purchase an asset from a kiosk. You can't specify which policy applies to a specific asset for the type when there is more than one policy attached. When someone meets the conditions that are easiest to meet, they are allowed to purchase and transfer the asset. 

Before you create a transfer policy, you can use the `quesryTransferpolicy` function to check the transfer policy associated with a type. This is similar to the `purchaseAndResolvePolicies` example.

Where is this example?

```typescript
import { createTransferPolicy } from '@mysten/kiosk';
import { TransactionBlock } from '@mysten/sui.js';

const createPolicyForType = async () => {
	const type = 'SomePackageId::type::MyType'; // the Type for which we're creating a Transfer Policy.
	const publisher = 'publisherObjectId'; // the publisher object id that you got when claiming the package that defines the Type.
	const address = 'AddressToReceiveTheCap';

	const tx = new TransactionBlock();
	// create transfer policy
	let transferPolicyCap = createTransferPolicy(tx, type, publisher);
	// transfer the Cap to the address.
	tx.transferObjects([transferPolicyCap], tx.pure(address, 'address'));

	// ... continue to sign and execute the transaction
	// ...
};
```

</details>

<details>

> <summary>Attach Rules (Royalty, Kiosk Lock) to the Transfer Policy</summary>

Reviewer: ```suggestion
<summary>Attach Royalty and Lock rules to a Transfer policy</summary>
```
